ambry-api/src/main/java/com.github.ambry/frontend/Operations.java

```diff
@@ -22,4 +27,10 @@
   public static final String UPDATE_TTL = "updateTtl";
   public static final String STITCH = "stitch";
   public static final String GET_CLUSTER_MAP_SNAPSHOT = "getClusterMapSnapshot";
+  public static final Set<String> OperationsSet;
+
+  static {
+    OperationsSet = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
I also remembered that we had some code in AmbryLI that did this same thing using reflection (so that we don't have to update the set every time we add a new operation):
```java
  static {
    INTERNAL_KEYS =
        Collections.unmodifiableList(getStaticFieldValuesAsStrings(InternalKeys.class).collect(Collectors.toList()));
    OPERATIONS =
        Collections.unmodifiableSet(getStaticFieldValuesAsStrings(Operations.class).collect(Collectors.toSet()));
  }

  /**
   * Get a stream of the accessible {@link String} values of the static fields in the provided {@link Class}.
   * @param type the {@link Class} to get static field values from.
   * @return a {@link Stream} of the static field value strings.
   */
  private static Stream<String> getStaticFieldValuesAsStrings(Class type) {
    return Arrays.stream(type.getFields()).filter(field -> Modifier.isStatic(field.getModifiers())).map(field -> {
      try {
        return field.get(null).toString();
      } catch (IllegalAccessException e) {
        throw new IllegalStateException("Could not get value of a static field, " + field + ", in " + type, e);
      }
    });
  }
```

Consider moving this over to open source.

Reply:
Good idea. This helper method is moved to `Utils.java` and invoked in `AmbrySecurityService`.